ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/JdbcLedgerDao.scala

```diff
@@ -315,88 +256,6 @@ private class JdbcLedgerDao(
-      ledgerEntries: Vector[(Offset, LedgerEntry)],
-      newLedgerEnd: Offset,
-  )(implicit loggingContext: LoggingContext): Future[Unit] = {
-    logger.info("Storing initial state")
-    dbDispatcher.executeSql(metrics.daml.index.db.storeInitialStateFromScenario) {
-      implicit connection =>
-        ledgerEntries.foreach { case (offset, entry) =>
-          entry match {
-            case tx: LedgerEntry.Transaction =>
-              val completionInfo = for {
-                actAs <- if (tx.actAs.isEmpty) None else Some(tx.actAs)
-                applicationId <- tx.applicationId
-                commandId <- tx.commandId
-                submissionId <- tx.submissionId
-              } yield state.CompletionInfo(
-                actAs,
-                applicationId,
-                commandId,
-                None,
-                Some(submissionId),
-                None, // TODO Ledger Metering
-              )
-
-              sequentialIndexer.store(
-                connection,
-                offset,
-                Some(
-                  state.Update.TransactionAccepted(
-                    optCompletionInfo = completionInfo,
-                    transactionMeta = state.TransactionMeta(
-                      ledgerEffectiveTime = tx.ledgerEffectiveTime,
-                      workflowId = tx.workflowId,
-                      submissionTime = null, // not used for DbDto generation
-                      submissionSeed = null, // not used for DbDto generation
-                      optUsedPackages = None, // not used for DbDto generation
-                      optNodeSeeds = None, // not used for DbDto generation
-                      optByKeyNodes = None, // not used for DbDto generation
-                    ),
-                    transaction = tx.transaction,
-                    transactionId = tx.transactionId,
-                    recordTime = tx.recordedAt,
-                    divulgedContracts = Nil,
-                    blindingInfo = None,
-                  )
-                ),
-              )
-            case LedgerEntry.Rejection(
-                  recordTime,
-                  commandId,
-                  applicationId,
-                  submissionId,
-                  actAs,
-                  reason,
-                ) =>
-              sequentialIndexer.store(
-                connection,
-                offset,
-                Some(
-                  state.Update.CommandRejected(
-                    recordTime = recordTime,
-                    completionInfo = state
-                      .CompletionInfo(
-                        actAs,
-                        applicationId,
-                        commandId,
-                        None,
-                        submissionId,
-                        None, // TODO Ledger Metering
-                      ),
-                    reasonTemplate = reason.toParticipantStateRejectionReason(errorFactories)(
```

Comment:
I think `toParticipantStateRejectionReason` can go as well

Reply:
There were some things that could be removed from there, indeed. But not all of them, since we still keep the `Rejection`s used by sandbox-classic, the `ActiveLedgerState` and friends around for the old Scala migrations. Removed what I could now. Follow-up clean-up PRs will take care of the rest when deemed possible.